projects/storefrontlib/src/cms-components/product/product-variants/guards/product-variant.guard.ts

```diff
@@ -22,13 +22,15 @@ export class ProductVariantGuard implements CanActivate {
   canActivate(): Observable<boolean | UrlTree> {
     return this.routingService.getRouterState().pipe(
       map(state => state.nextState.params.productCode),
-      filter(Boolean),
-      switchMap((productCode: string) =>
-        this.productService.get(productCode, ProductScope.VARIANTS)
-      ),
-      filter(Boolean),
+      switchMap((productCode: string) => {
+        if (productCode) {
```

Comment:
`if (Boolean(productCode)) {` might be cleaner, or at least there seems to be a convention in the code base to use the Boolean constructor whenever a non boolean var is used as a boolean.